docs/api.rst

```diff
@@ -348,10 +348,15 @@ Note that ``->>`` is used to compare ``blood_type`` as ``text``. To compare with
     { "id": 15, "age": 35 }
   ]
 
+
+.. note::
+
+   When using the ``->`` and ``->>`` operators, PostgREST uses a query like ``to_jsonb(<col>)->'field'``. To make filtering and ordering on those nested fields use an index, the index needs to be created on the same expression, including the ``to_jsonb(...)`` call.
```

Comment:
Aha, this is an important detail that I missed. How about:

```suggestion
.. important::

   When using the ``->`` and ``->>`` operators, PostgREST uses a query like ``to_jsonb(<col>)->'field'``. To make filtering and ordering on those nested fields use an index, the index needs to be created on the same expression, including the ``to_jsonb(...)`` call.
```

Reply:
Perhaps a snippet can be added to be more explicit? Like:

```sql
CREATE INDEX ON mytable((to_jsonb(data) -> 'identification' ->> 'registration_number'));
```

Reply:
> Aha, this is an important detail that I missed.

Yeah. The requirement to add an index on a expression was already the case before when using JSON fields. Now I wonder whether existing indexes for JSON fields need to be changed from `col->'field'`` expressions to `to_jsonb(col)->'field'`. I*m pretty sure the answer is yes. Would you consider this a breaking change?

Reply:
Yeah, I think it's a breaking change since it invalidates previous indexes, better to be clear about it.

Reply:
Hm, ok. So we either revert the commit in the postgrest repo to be able to release a minor release next - or we will make the next release a major release. In that case, I would go ahead an merge some of my other breaking changes that are still open.

Reply:
Major release sounds good. Ideally we'd have some patch/minor releases after a major one but reverting the commit seems off since we already made a pre-release.